File: python/908. Smallest Range I.py
class Solution(object):
    def smallestRangeI(self, A, K):
        """
        :type A: List[int]
        :type K: int
        :rtype: int
        """
        A.sort()
        diff = A[-1] - A[0]
        return min(diff, max(0,diff - 2*K))

File: go/src/leetcode/908/908. Smallest Range I.go
/**
@author ZhengHao Lou
Date    2022/04/30
*/
package main

import "math"

/**
https://leetcode-cn.com/problems/smallest-range-i/
*/
func smallestRangeI(nums []int, k int) int {
	var mi, mx = math.MaxInt32, math.MinInt32
	for i := range nums {
		mx = max(mx, nums[i])
		mi = min(mi, nums[i])
	}
	var s = mx - mi
	return min(s, max(s-k<<1, 0))
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}
